lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -427,18 +428,24 @@ private function generateMagicGet(ClassMetadata $class)
             if ($reflectionClass->getMethod('__get')->returnsReference()) {
                 $returnReference = '& ';
             }
-        }
+
+			$nameParameterTypeHint = $this->getMethodParameterTypeHintByReflection($reflectionClass, '__get', 'name');
```

Comment:
What happens if the parent class parameter name is not called `'name'`?

For instance:

```php
public function __get(string $n);
```

Consider using positional indexes for this API instead: that should rule out this scenario.

Reply:
Note: this needs an added test case

Reply:
Case `__get(string $n)` added and works.